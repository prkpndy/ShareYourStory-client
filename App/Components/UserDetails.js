import React from 'react';
import {View, Text, StyleSheet} from 'react-native';

const UserDetails = props => {
    return (
        <View style={styles.userInfo}>
            {Object.keys(props.userDetails).map((key, index) => {
                const value = props.userDetails[key];
                if (value) {
                    return <Text key={key}>value</Text>;
                } else {
                    return null;
                }
            })}
        </View>
    );
};

const styles = StyleSheet.create({
    userInfo: {
        alignItems: 'center',
    },
});

export default UserDetails;
