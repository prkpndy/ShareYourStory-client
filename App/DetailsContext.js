import React from 'react';

const DetailsContext = React.createContext();

export default DetailsContext;
