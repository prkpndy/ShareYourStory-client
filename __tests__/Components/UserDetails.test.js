/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import DetailsContext from '../../App/DetailsContext';
import UserDetails from '../../App/Components/UserDetails';

const userDetails = {
    name: 'John Doe',
    occupation: 'Front End Developer',
    email: 'johndoe@gmail.com',
    website: 'www.jest.js.io',
};

it('renders correctly', () => {
    const tree = renderer
        .create(
            <DetailsContext.Provider
                value={{
                    userDetails,
                }}>
                <UserDetails />
            </DetailsContext.Provider>,
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
