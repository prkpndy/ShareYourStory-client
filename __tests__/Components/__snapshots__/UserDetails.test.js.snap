// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Text>
    John Doe
  </Text>
  <Text>
    Front End Developer
  </Text>
  <Text>
    johndoe@gmail.com
  </Text>
  <Text>
    www.jest.js.io
  </Text>
</View>
`;
